source/fundamentals/logging.txt

```diff
@@ -0,0 +1,285 @@
+   
+      For more information on Logback, see the
+      `Logback manual <http://logback.qos.ch/manual/>`__.
+   
+   .. tab:: Log4j2
+      :tabid: Log4j2-binding
+   
+      If you add these dependencies to your ``pom.xml`` file, the MongoDB Java
+      driver uses the Log4j2 framework to output your logs.
+
+      .. code-block:: xml      
+      
+         <!-- SLF4J API -->
+         <dependency>
+           <groupId>org.slf4j</groupId>
+           <artifactId>slf4j-api</artifactId>
+           <version>1.7.30</version>
+         </dependency>
+         <!-- SLF4J-Log4j2 binding -->
+         <dependency>
+           <groupId>org.apache.logging.log4j</groupId>
+           <artifactId>log4j-slf4j-impl</artifactId>
+           <version>2.14.1</version>
+         </dependency>
+         <!-- Log4j2 core functionality -->
+         <dependency>
+           <groupId>org.apache.logging.log4j</groupId>
+           <artifactId>log4j-core</artifactId>
+           <version>2.14.1</version>
+         </dependency>
+
+      Once you have installed the above dependencies, log an error using the
+      following code.
+
+      .. code-block:: java
+
+         import org.slf4j.Logger;
+         import org.slf4j.LoggerFactory;
+
+         ...
+
+         Logger logger = LoggerFactory.getLogger(App.class);
+         logger.error("Logging an Error");
+
+      You should see output resembling the following:
+
+      .. code-block:: none
+
+         12:35:00.438 [main] ERROR <my package path> - Logging an Error
+
+      For more information on Log4j2, see the
+      `Log4j2 manual <https://logging.apache.org/log4j/2.x/manual/>`__.
+
+Configure Your Logger
+---------------------
+
+To configure your logger, you must configure the logging framework bound to
+SLF4J.
+
+The following subsection provides a concrete example showing how to configure
+your logger.
+
+Configure Your Log Level
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this subsection we show how to configure your logger's **log level**.
+
+A logger's log level specifies a lower bound for how urgent a message must be
+for the logger to output it.
+
+This example shows how to configure your logger's log level to INFO.
+Select the tab corresponding to the logging framework you are using in your
+project.
+
+.. tabs::
+
+   .. tab:: Logback
+      :tabid: logback-binding
+
+      Specify Logback configurations in a file named ``logback.xml``. Your
+      ``logback.xml`` file does not have to be in a specific location, but it must
+      be accessible from your classpath. 
+
+      The Logback framework defines the following log levels. You can
+      see the log levels listed below in order from most urgent to least
+      urgent: 
+      
+      - ERROR
+      - WARN
+      - INFO
+      - DEBUG
+      - TRACE 
+
+      To configure your Logback logger to log all messages of level INFO or
+      higher, set your ``logback.xml`` file to the following.
```

Comment:
This is repeating the same information "This example shows..." from line 191

Reply:
Thanks for the note. I thought this was helpful, but after starting at this for a while I can see how it is not adding significant value for readers. I changed this from stating what this section of code did, so providing a direction "set you file to this..." and then providing steps such that readers can produce a state to test their configuration. Fix:

Thanks for pointing this out. I am realizing how off base my understanding of repetition in writing is. Going forward, I will try and continue to develop my ear for when repetition is a good tool or whether something else would be better for learners.